### PR TITLE
Metadata update to support "Add to Steam" feature.

### PR DESCRIPTION
**Pull request by gablm**
Hii

This PR adds some assets and changes the preview metadata json file to reflect these additions.
It's still missing some assets, which I couldn't find on the internet:

- Banner (needs to be 600 x 900) [Example](https://github.com/CollapseLauncher/CollapseLauncher-ReleaseRepo/blob/d21c9ecfad7b0a4bf2121b0fe78153c70acf6af5/metadata/steam_assets/game_banners/banner_genshin.png)
  - StarRail: cn/bilibili
  - Genshin: cn/bilibili
  - Honkai: cn, kr, jp

- Preview (needs to be 920 x 430) [Example](https://github.com/CollapseLauncher/CollapseLauncher-ReleaseRepo/blob/d21c9ecfad7b0a4bf2121b0fe78153c70acf6af5/metadata/steam_assets/game_previews/preview_genshin.png)
  - StarRail: cn/bilibili
  - Honkai: cn, kr, jp

If anyone knows how to find or has any of the missing assets, please let me know.
All made of official art, with some taken from https://www.steamgriddb.com/

> Should only be merged with the release of the preview with the "Add To Steam" feature.
> Part of the https://github.com/CollapseLauncher/Collapse/issues/358 implementation process

### CODE REVIEW

**Comment by bagusnl** [comment]
Since this require changes in metadata, we will need to update our metadata internal tooling first as currently it doesn't have any builder for the necessary changes
CC @neon-nyan

**Comment by neon-nyan** [comment]
Hi @bagusnl and @gablm,

As per changes that's still in the works for this addition on the metadata level, I came up with how the property would be.
IMO, we can have the ``ZoneSteamAssets`` property to be a ``Dictionary<string, SteamGameProp>`` which each own properties (Logo, Hero, Banner and Preview) would be the key in the ``Dictionary`` then inside of ``SteamGameProp`` class, there would be ``URL`` and ``MD5`` property.

So basically, with how that come out. The usage would be something like this:
```CSharp
SteamGameProp steamGameLogoProp = ConfigV2Store[CurrentGameTitle][CurrentGameRegion].ZoneStreamAssets["Logo"];
// Do something with steamGameLogoProp.URL and steamGameLogoProp.MD5
...
```

What do you think about this approach?

**Comment by gablm** [comment]
Hi!
I think that your approach works out fine.

Gonna change main-repo PR to reflect the naming and dictionary changes.

**Comment by bagusnl** [comment]
Agreed with the change, it makes it much cleaner and better for us when we switch to different cdn as well

**Comment by bagusnl** [comment]
Everything is all good on CDN side, Thank you for the work! Go ahead and make the necessary changes in the main PR to support direct feed from main CDNs @gablm
